NetworkingSampleApp/NetworkingSampleApp/Scenes/Upload/UploadService.swift

```diff
@@ -7,17 +7,17 @@
 
 import Foundation
 import Networking
+// This import suppresses warning: Non-sendable type 'AsyncPublisher<AnyPublisher<UploadTask.State, Never>>' ...
+@preconcurrency import Combine
+
+final class UploadService: Sendable {
+    static let shared = UploadService()
 
-final class UploadService {
     private let uploadManager: UploadAPIManaging
 
     init(uploadManager: UploadAPIManaging = UploadAPIManager()) {
         self.uploadManager = uploadManager
     }
-
-    deinit {
-        uploadManager.invalidateSession(shouldFinishTasks: false)
-    }
```

Comment:
Since it's async now it just crashes the app, there is a whole [pitch](https://forums.swift.org/t/isolated-synchronous-deinit/58177) for async deinit, but it's not implemented yet.